### DISCUSSION TITLE
Not able to load tables from .sqlite database

### DISCUSSION THREAD

**Discussion by deepakpunia20** [sfu-db/connector-x#244] [Q&A] [answered]
Hi, 

When I try to load data from table 'rain' using connectox, it is not able to identify the it but I am successfully able to do so using sqlite3.
Could you please let me know reason for it. I am using version - "connectorx-0.2.1a1"

![image](https://user-images.githubusercontent.com/73750090/156236514-54f20615-78b9-4ef6-af93-4987a84c1fdd.png)

**Comment by wangxiaoying** [comment]
Hi, can you try the absolute path of the database file? Here is an example of the connection string: `sqlite:///home/user/path/test.db`

**Reply by deepakpunia20** [reply]
Hi Wang,

My jupyter notebook is running in windows 10 environment and .sqlite file is placed at same location, where notebook runs. So, ideally absolute path is not required. Though, I tried with absolute path as well but 'AssertionError' is thrown. .sqlite file don't have any password.

May be you would like to give a try ? Link to download file is - https://media.moa-info-tg.kenya.precisionag.org/rain.sqlite

![image](https://user-images.githubusercontent.com/73750090/156341666-a9ec349f-93f4-49d9-a3c6-626cf924586d.png)

**Reply by wangxiaoying** [reply]
Can you try `sqlite://C:\\...`? In my above example, only `/home/user/path/test.db` part is the absolute path.

**Reply by deepakpunia20** [reply]
Hi Wang,

I tried but still it doesn't work. Could you please try to download .sqlite DB from URL mentioned in my earlier update and give it a try ? I have already tried out all possible ways but none works like - sqlite:///C:\\...  **OR** sqlite:////C:\\...   **OR**  sqlite://////C:\\...

![image](https://user-images.githubusercontent.com/73750090/156427219-dec0955c-c80c-455a-98f7-cc348d9bd6bb.png)

**Reply by wangxiaoying** [reply]
Interesting, from the error message, it seems like it cannot recognize the file path correctly on windows. We will fix this issue. Thanks for reporting!

**Comment by wangxiaoying** [comment, marked as answer]
Hi @deepakpunia20 , can you try the newest alpha version `pip install connectorx==0.2.5-alpha.1` and see whether it works? You may also need to urlencode your file path like this:

```Python
import urllib

path = urllib.parse.quote("C:\\Users\\Deepak Punia\\rain.sqlite")
df = cx.read_sql(f"sqlite://{path}", query)
```

**Reply by deepakpunia20** [reply]
Hi Wang,

I installed version 0.2.5-alpha.1 but still no success.

![image](https://user-images.githubusercontent.com/73750090/156881546-9c778b29-0016-4f84-9db7-25eba9b9d197.png)

![image](https://user-images.githubusercontent.com/73750090/156881557-74b1b9eb-2f76-4f6e-9825-abb0c276f646.png)

![image](https://user-images.githubusercontent.com/73750090/156881567-0219d094-79a6-48dd-8eaf-901682c4c00a.png)

**Reply by wangxiaoying** [reply]
It seems like the file can be found and it's another problem now. It's weird since I can run the exact same query on linux:
<img width="793" alt="image" src="https://user-images.githubusercontent.com/5569610/156911547-27f6fb8b-827c-49cf-a513-1949c999224e.png">

Can you print the log by setting `RUST_LOG` (like [here](https://sfu-db.github.io/connector-x/freq_questions.html#how-to-print-log-in-python)) so we could narrow down the problem?

**Reply by deepakpunia20** [reply]
Hi Wang,

Below is the RUST_LOG:

![image](https://user-images.githubusercontent.com/73750090/157247348-c9e4975f-08d3-4b2e-ab8d-968494f67f2b.png)

**Reply by wangxiaoying** [reply]
Hi @deepakpunia20 , I think you didn't add the "sqlite://" at the beginning of the connection url. So it throws error directly when parsing the input paramters.

**Reply by deepakpunia20** [reply]
Hi @wangxiaoying , 

Sorry, Please find below the correct screenshot:

![image](https://user-images.githubusercontent.com/73750090/157298526-d7c3b159-a21e-43c8-91d4-603bfc0fa5f3.png)

**Reply by wangxiaoying** [reply]
Hi @deepakpunia20 , thanks for the quick response.

In order to print the log, you need to set the environment variable before importing connectorx exactly like the [example](https://sfu-db.github.io/connector-x/freq_questions.html#how-to-print-log-in-python). We will add this clarification to the webpage.

**Reply by deepakpunia20** [reply]
Hi @wangxiaoying ,

![image](https://user-images.githubusercontent.com/73750090/157299604-73e3e9fc-d61e-49ef-843e-b51c84f1e5e4.png)

**Reply by wangxiaoying** [reply]
Hi @deepakpunia20 , I still don't see any log printed out. Have you try to restart your notebook kernel? I'm suspecting that connectorx was not reloaded.

**Reply by deepakpunia20** [reply]
Hi @wangxiaoying ,

I tried in new notebook but still, I don't see any change in the result.

![image](https://user-images.githubusercontent.com/73750090/157301405-683e9854-2104-411d-a0d8-7a2bdbe384ca.png)

**Reply by wangxiaoying** [reply]
That's weird. I will try to find a windows machine and see whether I can reproduce the error.

**Reply by wangxiaoying** [reply]
Hi @deepakpunia20 , I tried on a windows machine and still cannot reproduce your error. Here is the result with log printed:
<img width="1110" alt="image" src="https://user-images.githubusercontent.com/5569610/157333096-7b802eea-6f27-44a7-af9e-4ab09c1aa00e.png">
with this script:
<img width="610" alt="image" src="https://user-images.githubusercontent.com/5569610/157333207-5ebf7838-fd9c-4a78-b3a5-da99071c5aa4.png">


Can you make sure that the connectorx used in your notebook is the correct version (0.2.5a1)?

**Reply by deepakpunia20** [reply]
Hi @wangxiaoying ,

Could you please paste query that you ran. I will paste it as it is and try to run it. I am using correct version- 

![image](https://user-images.githubusercontent.com/73750090/157370962-fb6fde01-1949-4f88-9627-486f7142b2af.png)

**Reply by deepakpunia20** [reply]
Hi @wangxiaoying ,

I tried to ran it from command prompt and I can see below log now:

C:\Users\Deepak Punia>python .\test_sqlite.py
sqlite://C%3A%5CUsers%5Crain.sqlite
[2022-03-09T04:21:27Z DEBUG connectorx::sources::sqlite] decoded conn: C:\Users\rain.sqlite
[2022-03-09T04:21:27Z DEBUG connectorx_python::pandas] Running dispatcher
[2022-03-09T04:21:27Z DEBUG connectorx::dispatcher] Fetching metadata
[2022-03-09T04:21:27Z DEBUG connectorx::sql] Transformed limit 1 query: SELECT count(*) FROM rain LIMIT 1
[2022-03-09T04:21:30Z DEBUG connectorx::dispatcher] Try get row rounts for entire result
[2022-03-09T04:21:30Z DEBUG connectorx::dispatcher] Manually count rows of each partitioned query and sum up
[2022-03-09T04:21:30Z DEBUG connectorx::sql] Transformed count query: SELECT count(*) FROM (SELECT count(*) FROM rain) AS CXTMPTAB_COUNT
[2022-03-09T04:21:32Z DEBUG connectorx::dispatcher] Allocate destination memory: 1x1
[2022-03-09T04:21:32Z DEBUG connectorx::dispatcher] Create destination partition
[2022-03-09T04:21:32Z DEBUG connectorx::dispatcher] Start writing
[2022-03-09T04:21:34Z DEBUG connectorx::dispatcher] Finalize partition 0
[2022-03-09T04:21:34Z DEBUG connectorx::dispatcher] Partition 0 finished
[2022-03-09T04:21:34Z DEBUG connectorx::dispatcher] Writing finished
Traceback (most recent call last):
  File ".\test_sqlite.py", line 12, in <module>
    df    = cx.read_sql(url, query)
  File "C:\Users\Deepak_Punia\anaconda3\lib\site-packages\connectorx\__init__.py", line 125, in read_sql
    df = reconstruct_pandas(result)
  File "C:\Users\Deepak_Punia\anaconda3\lib\site-packages\connectorx\__init__.py", line 210, in reconstruct_pandas
    placement=binfo.cids[0],
  File "C:\Users\Deepak_Punia\anaconda3\lib\site-packages\pandas\core\internals\blocks.py", line 3041, in make_block
    return klass(values, ndim=ndim, placement=placement)
  File "C:\Users\Deepak_Punia\anaconda3\lib\site-packages\pandas\core\internals\blocks.py", line 1727, in __init__
    super().__init__(values, placement, ndim)
  File "C:\Users\Deepak_Punia\anaconda3\lib\site-packages\pandas\core\internals\blocks.py", line 1608, in __init__
    placement = libinternals.BlockPlacement(placement)
  File "pandas\_libs\internals.pyx", line 45, in pandas._libs.internals.BlockPlacement.__init__
AssertionError

**Reply by deepakpunia20** [reply]
Hi @wangxiaoying ,

Looking at above log, it looked that connectorx version is not compatible with my pandas. I installed latest version of pandas and now, my query is running fine. I appreciate your prompt response to sort this issue.

![image](https://user-images.githubusercontent.com/73750090/157373051-001b7146-255b-420f-9ad1-1f0b7ef78d63.png)
